### PR TITLE
Cleanup and fix pkgconfig files so forge packages can be installed

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The purpose here is to fix https://github.com/conda-forge/octave-feedstock/issues/50 by fixing paths in the `.pc` file.

### CODE REVIEW

**Comment by ngam** [comment]
@conda-forge-admin, please rerender

(and make some noise!)

**Comment by izahn** [comment]
I'm not entirely sure I was on the right track here -- it turns out that packages install OK on linux as-is, and not on osx even with the changes here.

**Review by ocefpaf — APPROVED** [review]
LGTM but I'll wait for @isuruf b/c he knows way more than me about... Well, everything.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,6 +38,8 @@ requirements:
     - {{ cdt('libx11-devel') }}  # [linux]
     - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
   host:
+    - libgomp     # [linux]
+    - llvm-openmp  # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:42]: These should be in `build`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,10 +9,8 @@ source:
   sha256: 232065f3a72fc3013fe9f17f429a3df69d672c1f6b6077029a31c8f3cd58a66e
 
 build:
-  number: 1
+  number: 2
   skip: true  # [win]
-  binary_has_prefix_files:   # [unix]
-    - bin/octave-{{ version }}  # [linux]
```
Comment (isuruf) [inline recipe/meta.yaml:15]: Are you sure the octave gui works without this?

File: recipe/build.sh [hunk 3]
```diff
@@ -19,9 +19,20 @@ chmod +x build-aux/mk-opts.pl
...
+	export LD_FLAGS+=" -lobjc"
+fi
+
 make -j${CPU_COUNT}
 make install
 
+## Tried conda_build_config.yaml replacement from
+## https://github.com/conda-forge/vtk-feedstock/blob/master/recipe/conda_build_config.yaml
+## but it didn't work so here we use sed to fix up the .pc file
+sed -i "s|/home.*_build_env|${PREFIX}|g" ${PREFIX}/lib/pkgconfig/octave.pc
```
Comment (isuruf) [inline recipe/build.sh:33]:
```suggestion
sed -i "s|${BUILD_PREFIX}|${PREFIX}|g" ${PREFIX}/lib/pkgconfig/octave.pc
```

File: recipe/build.sh [hunk 4]
```diff
@@ -19,9 +19,20 @@ chmod +x build-aux/mk-opts.pl
     --with-hdf5-includedir=${PREFIX}/include \
     --with-hdf5-libdir=${PREFIX}/lib
 
+## see https://trac.macports.org/ticket/61865 and https://github.com/macports/macports-ports/commit/be633a5c854958eb13e6b927228613858acddb4b
+if [ "$(uname)" == "Darwin" ]; then
```
Comment (isuruf) [inline recipe/build.sh:23]:
```suggestion
if [[ "${target_platform}" == "osx"-* ]]; then
```

File: recipe/meta.yaml (hunk 1 above)
Comment (izahn) [reply recipe/meta.yaml:42]: Yeah, this was just me flailing around.

File: recipe/build.sh (hunk 4 above)
Comment (izahn) [reply recipe/build.sh:23]: Use target_platform

File: recipe/meta.yaml (hunk 2 above)
Comment (izahn) [reply recipe/meta.yaml:15]: It does work on CentOS7 at work, but I just tested again and it crashes on Archliinux at home. I'll change this back too.

File: recipe/meta.yaml (hunk 1 above)
Comment (izahn) [reply recipe/meta.yaml]: Fixed in https://github.com/conda-forge/octave-feedstock/pull/55/commits/a993b9a63dabd6e4713f172969e17748c4c470c0

File: recipe/meta.yaml (hunk 2 above)
Comment (izahn) [reply recipe/meta.yaml]: fixed in https://github.com/conda-forge/octave-feedstock/pull/55/commits/a993b9a63dabd6e4713f172969e17748c4c470c0

**Comment by izahn** [comment]
@conda-forge-admin please rerender

**Pull request closed without merging**